pkg/middleware/headers.go

```diff
@@ -44,7 +44,11 @@ func newStripHeaders(headers []options.Header) alice.Constructor {
 func flattenHeaders(headers http.Header) {
 	for name, values := range headers {
 		if len(values) > 1 {
-			headers.Set(name, strings.Join(values, ","))
+			if name == "Set-Cookie" {
+				// omit flattening Set-Cookie Headers.
+			} else {
+				headers.Set(name, strings.Join(values, ","))
+			}
```

Comment:
How about inlining the secondary if clause, I don't think we need the extra if else.
The comment could go above.

Also, I think it's best to state why we aren't flattening rather than what we are doing here, could we update the comment to explain the motivation why, else I don't think the comment adds a lot right now

Reply:
Squashed the additional if-else, added further comment.